doc/mysqlnd_azure_log.md

```diff
@@ -0,0 +1,124 @@
+/etc/apache2/sites-available/000-default.conf	| ErrorLog	| logOutput1 will go here if the error_log is not set in php.ini and www.conf. <br/> Please notice that the logs for single request is in single line since fastcgi treat one request’s log as single line)
+/etc/apache2/apache2.conf	| ErrorLog	| Apache’s global error log path. <br/> logOutput1 will go here if the error_log is not set in php.ini and www.conf and ErrorLog is not set in each virtual instance
+
+Work scenario with Nginx is similar.
+
+ ### mysqlnd\_azure.logfilePath
+
+Name | mysqlnd\_azure.logfilePath
+:----- | :------
+Comment | Filename that you want your log writes to, if mysqlnd\_azure.logOutput=2
+Type | String
+Accepted Value | String length <= 255, legal filename string
+Default | mysqlnd\_azure\_runtime.log
+Dynamic | NO
+Note | Filename length cannot exceed the file system's restriction(255), otherwise system will use the default filename and throw a warning "Given logfile name too long, redirected to default: mysqlnd_azure_runtime.log"
+
+ ### mysqlnd\_azure.logLevel
+Name | mysqlnd\_azure.logLevel
+:----- | :------
+Comment | The verbose level of log we generate.
+Type | Enumeration
+Accepted Value | [ 0 \| 1 \| 2 \| 3 ]
+Default | 0 (OFF)
+Dynamic | Yes
+Note | This variable is dynamic, you may use user srcipt API `ini_set('mysqlnd_azure.logLevel', '$VAL'); ` to check log with different verbose level.
+
+### Loglevel supported
+we support 3 level of logs for user:
+- [DEBUG]:
+  1. source code path(inside which function);
+  2. connection information(except password);
+  3. verbose running status (e.g. customer given redirected server information directly... )
+- [INFO]:
+  1. connection fallback when mysqlnd\_azure.enableRedirection=preferred with details (cannot get redirection
+      info from OK packet/ try connect using redirect information failed , ...);
+  2. cache operations
+- [ERROR]:
+  1. any operation may cause customer see connection break (e.x. original conn/ redirected\_conn
+    established failed, ... and connection fall back when mysqlnd\_azure.enableRedirection = ON)
+
+> [SYSTM] maked log rows for Azure mysqlnd Log Module itself, e.g. invalid file length:
+> ```
+> 2020-03-27 09:09:05 [SYSTM] Given logfile name too long, redirected to default: mysqlnd_azure_runtime.log
+> ```
+
+> note: OOM not included this version.
+
+#### Level value relationship to level logged
+- 0: No log written
+- 1: [SYSTM] + [ERROR]
+- 2: [SYSTM] + [ERROR] + [INFO]
+- 3: [SYSTM] + [ERROR] + [INFO] + [DEBUG]
+
+
+## Usage Example
+> You can add to section [mysqlnd\_azure] in file `php.ini`
```

Comment:
```suggestion
> You can add to section [mysqlnd\_azure] in file `php.ini`. In the below example, the logs are output to a pre-defined file (using logOutput = 2) called "test.log" (using logfilePath). The logs level output includes system, error, info, and debug logs (set using logLevel = 3).

the logs are output to a pre-defined file called "test.log".  
```

Reply:
I have updated the test setting a bit to use absolute file path which is more recommended. Please help check.